lib/spack/spack/cmd/config.py

```diff
@@ -431,6 +444,92 @@ def config_revert(args):
         tty.msg(msg.format(cfg_file))
 
 
+def config_prefer_upstream(args):
+    """Generate a packages config based on the configuration of all upstream
+    installs."""
+
+    scope = args.scope
+    if scope is None:
+        scope = spack.config.default_modify_scope('packages')
+
+    specs = spack.store.db.query(installed=[InstallStatuses.INSTALLED])
```

Comment:
This will avoid the need to import from the database
```suggestion
    specs = spack.store.db.query(installed=True)
```